docs/core/entity.md

```diff
@@ -142,6 +142,33 @@ entity class for details.
 If an integration needs to access its own properties it should access the property (`self.name`), not the class or instance attribute (`self._attr_name`).
 :::
 
+### Example
+
+The below code snippet gives an example of best practices for when to implement property functions, and when to use class or instance attributes.
+
+```py
+class SomeEntity():
+    _attr_device_clas = DEVICE_CLASS_TEMPERATURE  # This will be common to all instances of SomeEntity
+    def __init__():
+        self._attr_available = False  # This overrides the default
+        self._attr_name = device.get_friendly_name()
+
+        # The following should be avoided:
+        if some_complex_condition and some_other_condition and something_is_none_and_only_valid_after_update and device_available:
+           ...
+
+    def update()
+        if self.available  # Read current state, no need to prefix with _attr_
+            # Update the entity
+            device_update()
+
+        if error:
+            self._attr_available = False  # Set property value
+            return
+        # We don't need to check if device available here
+        self._attr_is_on = device.get_state()  # Update "is_on" property
```

Comment:
```suggestion
        self._attr_is_on = self._device.get_state()  # Update "is_on" property
```

Reply:
I think comments suggested by @cgtobi are ok, not mandatory, but so that we don't show examples which does not run they should be added. Besides that it looks ok